### PR TITLE
Feature/switch agent init

### PR DESCRIPTION
**Pull request by manghuli**
# Summary of Changes

switched hooks package to npm version and new config initialization

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

This is just a reminder about the most common mistakes. Please make sure that you tick all _appropriate_ boxes. But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this).
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components.
- [ ] Run prettier: `npm run style-format`
- [ ] Updated **documentation** for changed code and new or modified features.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: assets/images.js [hunk 1]
```diff
@@ -1,22 +1,5 @@
 const images = {
-  arrow: require('./img/arrow.png'),
-  arrowDown: require('./img/arrow-down.png'),
-  checkmark: require('./img/checkmark.png'),
-  infoBlue: require('./img/info-blue.png'),
-  infoGray: require('./img/info-gray.png'),
-  infoWhite: require('./img/info-white.png'),
-  logo: require('./img/logo.png'),
```
Comment (TimoGlastra) [inline assets/images.js:8]: Why are all these images removed. Are they not needed anymore?

**Review by TimoGlastra — DISMISSED** [review]
LGTM 👍

File: assets/images.js (hunk 1 above)
Comment (manghuli) [reply assets/images.js:8]: ya, they were being used as icons but now we're using the materialIcons

**Comment by JamesKEbert** [comment]
@NeilSMyers did you run into any issues with usage in Android or iOS with the upgrade to RN 66? Specifically when receiving a credential too.

@TimoGlastra note that we've created a directory for genesis files vs downloading them at runtime. Is this along the lines of what you were proposing, or not quite?

**Comment by TimoGlastra** [comment]
@JamesKEbert yes that's exactly what I meant.

If we update to the latest AFJ version we can now also use multiple ledgers!

File: App/screens/ProofRequest.tsx [hunk 2]
```diff
@@ -46,13 +46,18 @@ const CredentialOffer: React.FC<Props> = ({ navigation, route }) => {
   }, [proof])
 
   const getRetrievedCredentials = async () => {
+<<<<<<< HEAD
```
Comment (cvarjao) [inline App/screens/ProofRequest.tsx]: is this an unresolved conflict?

Comment (manghuli) [reply App/screens/ProofRequest.tsx]: oopsies, that's strange. fixed it

**Comment by jleach** [comment]
Will this be an issue?
```
➜  ios git:(main) ✗ pod install
Auto-linking React Native modules for target `AriesBifold`: RNCAsyncStorage, RNCMaskedView, RNFS, RNGestureHandler, RNKeychain, RNReanimated, RNSVG, RNScreens, RNVectorIcons, indy-sdk-react-native, react-native-camera, react-native-config, react-native-get-random-values, and react-native-safe-area-context
Analyzing dependencies
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
[!] CocoaPods could not find compatible versions for pod "RCT-Folly":
  In snapshot (Podfile.lock):
    RCT-Folly (from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`)

  In Podfile:
    RCT-Folly (from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`)

It seems like you've changed the version of the dependency `RCT-Folly` and it differs from the version stored in `Pods/Local Podspecs`.
You should run `pod update RCT-Folly --no-repo-update` to apply changes made locally.
```

After merging this into my local `main` branch I deleted `node_modules` and ran `npm i` then `pod install`.

**Comment by manghuli** [comment]
> Will this be an issue?
> 
> ```
> ➜  ios git:(main) ✗ pod install
> Auto-linking React Native modules for target `AriesBifold`: RNCAsyncStorage, RNCMaskedView, RNFS, RNGestureHandler, RNKeychain, RNReanimated, RNSVG, RNScreens, RNVectorIcons, indy-sdk-react-native, react-native-camera, react-native-config, react-native-get-random-values, and react-native-safe-area-context
> Analyzing dependencies
> Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
> [!] CocoaPods could not find compatible versions for pod "RCT-Folly":
>   In snapshot (Podfile.lock):
>     RCT-Folly (from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`)
> 
>   In Podfile:
>     RCT-Folly (from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`)
> 
> It seems like you've changed the version of the dependency `RCT-Folly` and it differs from the version stored in `Pods/Local Podspecs`.
> You should run `pod update RCT-Folly --no-repo-update` to apply changes made locally.
> ```
> 
> After merging this into my local `main` branch I deleted `node_modules` and ran `npm i` then `pod install`.

ya, that happens, either delete the RCT-Folly pod and reinstall, or run that command they tell you in the error

**Comment by jleach** [comment]
@NeilSMyers Should all the hooks come from `react-hooks` now? Some files, like `CardDetails.tsx` still import from `aries-hooks` but the file `genesis-utils` was deleted from there yet the index file still imports it. Causes an issue:

**Comment by manghuli** [comment]
> @NeilSMyers Should all the hooks come from `react-hooks` now? Some files, like `CardDetails.tsx` still import from `aries-hooks` but the file `genesis-utils` was deleted from there. Causes an issue:
> 
> ```
> error Unable to resolve module aries-hooks from /Users/jleach/Workspace/bcgov/aries-mobile-agent-react-native/App/screens/ContactDetails.tsx: aries-hooks could not be found within the project or in these directories:
>   node_modules
> ```

Ya, this PR will switch all the imports, CredentialDetails isn't in use yet so I'll fix that up when we decide what that screen is going to look like.

**Comment by JamesKEbert** [comment]
> CredentialDetails isn't in use yet so I'll fix that up when we decide what that screen is going to look like.

@NeilSMyers, perhaps either commenting out & adding a #todo or fixing from the top level would be good here to prevent issues/confusion?

**Comment by manghuli** [comment]
> > CredentialDetails isn't in use yet so I'll fix that up when we decide what that screen is going to look like.
> 
> @NeilSMyers, perhaps either commenting out & adding a #todo or fixing from the top level would be good here to prevent issues/confusion?

@JamesKEbert  @jleach I just went ahead and switched the import

**Comment by jleach** [comment]
> ya, that happens, either delete the RCT-Folly pod and reinstall, or run that command they tell you in the error

@NeilSMyers The old version is coached which is causing the iOS build fail - which is ok. Just bump the `cacheId` in the [main.yaml](https://github.com/hyperledger/aries-mobile-agent-react-native/blob/main/.github/workflows/main.yaml)  workflow and a fresh cache will be created. In GitHub Actions one can't delete a cache, we can just create new ones and let the old ones expire and get cleaned up.

```yaml
env:
  cacheId: '3' # increment to expire the cache
```

**Review by amanji — APPROVED** [review]
The only thing I would comment on is that, perhaps we should plan to introduce handling for the case when initializing an agent fails (unless this is already in place) and the agent is therefore `undefined`.

**Comment by manghuli** [comment]
> The only thing I would comment on is that, perhaps we should plan to introduce handling for the case when initializing an agent fails (unless this is already in place) and the agent is therefore `undefined`.

The useAgent hook returns a loading value that only switches to false when the agent is initialized

**Pull request merged**